packages/api/src/valueObjects/ValueObject.ts

```diff
@@ -0,0 +1,26 @@
+export abstract class ValueObject {
+    private identifier: string;
+    private static identifierName: string;
+
+    constructor(identifier: string) {
+        this.identifier = identifier;
+    }
+
+    abstract get name(): string;
+
+    static getName() {
+        return this.identifierName;
+    }
+
+    get value() {
+        return this.identifier;
+    }
+
+    toString() {
+        return this.value;
+    }
+
```

Comment:
J'aurais ajouté 
```suggestion
    abstract static isValid(identifier: string | null | undefined): boolean;
```
pour regrouper tous les `isSiret`, `isRidet` etc et simplifier le code quand on cherche à savoir quel type d'identifiant c'est.

Reply:
du coup pour l'instant faut de mieux chaque classe définie elle même ses classes statiques